File: lib/domain/preferences/chart_aggregation_mode.dart
/// How the intraday vitals charts summarise their data.
enum ChartAggregationMode {
  off,
  min5,
  min10,
  min30;

  /// Bucket width in minutes, or null when [off] (raw data).
  int? get bucketMinutes => switch (this) {
        ChartAggregationMode.off => null,
        ChartAggregationMode.min5 => 5,
        ChartAggregationMode.min10 => 10,
        ChartAggregationMode.min30 => 30,
      };
}

File: app/src/main/kotlin/tech/mmarca/openvitals/domain/preferences/ChartAggregationMode.kt
package tech.mmarca.openvitals.domain.preferences

/** How the intraday vitals charts summarise their data. */
enum class ChartAggregationMode {
    OFF,
    MIN5,
    MIN10,
    MIN30,
    ;

    /** Bucket width in minutes, or null when [OFF] (raw data). */
    val bucketMinutes: Int?
        get() = when (this) {
            OFF -> null
            MIN5 -> 5
            MIN10 -> 10
            MIN30 -> 30
        }
}
